src/components/Navbar/Navbar.tsx

```diff
@@ -11,13 +11,43 @@ import { selectNavbar } from '@/redux/slices/navbar';
 
 const Navbar = () => {
   const { isActive } = useOnboarding();
-  const { isVisible: isNavbarVisible } = useSelector(selectNavbar, shallowEqual);
+  const {
+    isVisible: isNavbarVisible,
+    isSettingsDrawerOpen,
+    isNavigationDrawerOpen,
+    isSearchDrawerOpen,
+  } = useSelector(selectNavbar, shallowEqual);
   const showNavbar = isNavbarVisible || isActive;
 
+  useEffect(() => {
```

Comment:
Could you please extract all of the useEffect logic to a custom hook and just use the custom hook in the `Navbar` component.

One more thing regarding perfomance, is to add throttling to the `resize` event listener function.

```
function throttle(fn, time) {
    let timeout = null;
    return function () {
        if(timeout) return;
        const context = this;
        const args = arguments;
        const later = () => {
            fn.call(context, ...args);
            timeout = null;
        }
        timeout = setTimeout(later, time);
    }
}

const handleResizeThrottled = throttle(setScrollbarWidth, 500);
```